### ISSUE TITLE
File.separatorChar causing problems: using "/" to reference templates on Windows causes LoaderException: "Could not find template"

### ISSUE THREAD

**Issue by atsu85** [mbosecke/pebble#162]
### Templates
templates/f1/sub.peb

tries to extend

templates/f2/base.peb as shown bellow:
```
{% extends "../f2/base.peb" %}
```

### Problem
When trying to evaluate template smth like this (pseudocode):
``` Java
			Path filePath = ...;
			PebbleEngine engine = new PebbleEngine.Builder().build();
			PebbleTemplate template = engine.getTemplate(filePath.toString());
...
Map<String, Object> context = ...
Writer writer = ...
template.evaluate(writer, context);
```
**I get an exception on Windows** (but not on Linux):
```
Caused by: com.mitchellbosecke.pebble.error.LoaderException: Could not find template "f2/base.peb" (?:?)
        at com.mitchellbosecke.pebble.loader.DelegatingLoader.getReader(DelLoader.java:74) ~[pebble-2.0.0.jar:na]
        at com.mitchellbosecke.pebble.loader.DelegatingLoader.getReader(DelegatingLoader.java:27) ~[pebble-2.0.0.jar:na]
        at com.mitchellbosecke.pebble.PebbleEngine.retrieveReaderFromLoader(PebbleEngine.java:173) ~[pebble-2.0.0.jar:na]
        at com.mitchellbosecke.pebble.PebbleEngine.access$300(PebbleEngine.java:52) ~[pebble-2.0.0.jar:na]
        at com.mitchellbosecke.pebble.PebbleEngine$1.call(PebbleEngine.java:126) ~[pebble-2.0.0.jar:na]
        at com.mitchellbosecke.pebble.PebbleEngine$1.call(PebbleEngine.java:120) ~[pebble-2.0.0.jar:na]
        at com.google.common.cache.LocalCache$LocalManualCache$1.load(LocalCache.java:4742) ~[guava-18.0.jar:na]
        at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3527) ~[guava-18.0.jar:na]
        at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2319) ~[guava-18.0.jar:na]
        at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2282) ~[guava-18.0.jar:na]
        at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2197) ~[guava-18.0.jar:na]
        at com.google.common.cache.LocalCache.get(LocalCache.java:3937) ~[guava-18.0.jar:na]
        at com.google.common.cache.LocalCache$LocalManualCache.get(LocalCache.java:4739) ~[guava-18.0.jar:na]
        at com.mitchellbosecke.pebble.PebbleEngine.getTemplate(PebbleEngine.java:120) ~[pebble-2.0.0.jar:na]
        at com.mitchellbosecke.pebble.template.PebbleTemplateImpl.setParent(PebbleTemplateImpl.java:329) ~[pebble-2.0.0.jar:na]
        at com.mitchellbosecke.pebble.node.ExtendsNode.render(ExtendsNode.java:32) ~[pebble-2.0.0.jar:na]
        at com.mitchellbosecke.pebble.node.BodyNode.render(BodyNode.java:46) ~[pebble-2.0.0.jar:na]
        at com.mitchellbosecke.pebble.node.RootNode.render(RootNode.java:32) ~[pebble-2.0.0.jar:na]
        at com.mitchellbosecke.pebble.template.PebbleTemplateImpl.evaluate(PebbleTemplateImpl.java:106) ~[pebble-2.0.0.jar:na]
        at com.mitchellbosecke.pebble.template.PebbleTemplateImpl.evaluate(PebbleTemplateImpl.java:84) ~[pebble-2.0.0.jar:na]
```

**Comment by atsu85** [comment]
problem is probably related to using default template loader (that delegates to FileLoader), and the fact that `File.separatorChar` is only accepted as path separator:
https://github.com/mbosecke/pebble/blob/master/src/main/java/com/mitchellbosecke/pebble/loader/FileLoader.java#L57-L78

**Comment by atsu85** [comment]
i got it working by subclassing FileLoader:
``` Java
	@Override
	public String resolveRelativePath(String relativePath, String anchorPath) {
		String osRelativePath = relativePath;
		if(File.separatorChar == '\\') {
			// OS is Windows, Windows file names can't contain '/'
			osRelativePath = relativePath.replace('/', '\\');
		}
		return super.resolveRelativePath(osRelativePath, anchorPath);
	}
```

**Comment by mbosecke** [comment]
Yup, this sounds like a bug. Relative template paths are a fairly new feature. 

I'm guessing the core of the problem lies within [PathUtils](https://github.com/mbosecke/pebble/blob/master/src/main/java/com/mitchellbosecke/pebble/utils/PathUtils.java). It looks like it assumes that the separators used in the `{% extends "../f2/base.peb" %}` expression will be the same separators used to load the current template. I'm guessing that if you were to write `{% extends "..\f2\base.peb" %}` it will work on windows but not on linux.

Ideally, we should be able to use either separator in our templates and the templates will work across both platforms.

**Comment by atsu85** [comment]
> I'm guessing the core of the problem lies within PathUtils

I also looked at it when i discovered [PathUtils.resolveRelativePath(relativePath, anchorPath) call from FileLoader.resolveRelativePath](https://github.com/mbosecke/pebble/blob/master/src/main/java/com/mitchellbosecke/pebble/loader/FileLoader.java#L134)

But when i moved the change from FileLoader.resolveRelativePath to the beginning PathUtils.resolveRelativePath, then lots of tests broke - so i decided to do the minimal change for my project (I didn't have time to investigate more and send the proper pull request with test and fix).

**Comment by mbosecke** [comment]
I believe I've fixed the issue. If you get an opportunity to test my latest commits to see if it resolves it for you, that would be great. Thanks.

**Comment by atsu85** [comment]
Thanks @mbosecke, confirming that this issue is fixed in v2.1.0 :)
Thanks for quick commit :)
